lib/dom-parser.js

```diff
@@ -243,6 +248,11 @@ DOMHandler.prototype = {
 					this.doc.doctype = dt;
 	    }
 	},
+
+	internalEntityDecl:function(name, value) {
+		this.entityMap[name] = value;
```

Comment:
Just thinking out loud without checking any spec: Is it reasonable to allow overriding existing declarations?

Reply:
Good question and spec reference! Since the default `entityMap`s defined in `lib/entities.js` are frozen (using `Object.freeze` from `lib/conventions.js`), this kind of works right now; but only the predefined entities, not if repeating a declaration.

So I added an explicit check and redeclaration warning in 311ffc3cd178960952dc23a8986babe6bfd7934f.

The prototype is defined on the type, not in the prototype, so an undefined `'prototype'` key returns `undefined` (and a `&prototype;` entity can be defined in an entity declaration).

Reply:
Also, the prototypical inheritance is an old manoeuvre which ought to be efficient. But at least moving it out to a utility could  be wise. I did consider using `Object.assign` here, but AFAIK that is an ES6 feature. (If a future overhaul decides to use e.g. `hasOwnProperty` tests, or replace the entityMap with an ES6 `Map` here, this would of course have to be reworked.)

Reply:
> Since the default entityMaps defined in lib/entities.js are frozen (using Object.freeze from lib/conventions.js)

Please be aware it will not be frozen in a runtime where the `Object.freeze` method is not available. So thank you for adding that check.

I think it would be good to add a test for that, and also the prototype entity, just to be sure it doesn't break in the future.

Reply:
> Also, the prototypical inheritance is an old manoeuvre which ought to be efficient. But at least moving it out to a utility could be wise. I did consider using `Object.assign` here, but AFAIK that is an ES6 feature. (If a future overhaul decides to use e.g. `hasOwnProperty` tests, or replace the entityMap with an ES6 `Map` here, this would of course have to be reworked.)

I just [played with this approach a bit](https://stackblitz.com/edit/js-xkevsc) and after checking how the `entityMap` is currently used to replace entities filed the bug #370.
So once that bug is fixed your current solution no longer works.

I assume we will not get around either copying the entities. (I wouldn't want to touch the sax parser more then absolutely required right now, since we have plans to get rid of it: #55.)

Reply:
The mentioned bug was now fixed and the fix was released as 0.8.1.

Please update your branch (I can also do that if you prefer), after which I assume some tests would fail. The solution should be to copy the entitymap.
Since we can not just rely on `Object.assign` I will provide something similar as part of `lib/conventions.js` in the next days (in a separate PR).

Reply:
Hans been done as part of #379